install/helm/agones/templates/crds/fleet.yaml

```diff
@@ -128,7 +128,7 @@ spec:
                         description: The name of the Counter or List. If not found on the GameServer, those GameServer with the key will have priority over those that do not.
                       order:
                         type: string
-                        description: Ascending or Descending sort order. Default is "Ascending" so remove smaller total capacity first. "Descending" would remove larger total capacity first.
+                        description: Ascending or Descending sort order. Default is "Ascending" so remove smaller available capacity first. "Descending" would remove larger available capacity first.
```

Comment:
Huh! Interesting! So (as you stated below) -- I can't remember the history - was there a specific reason for `priorities` to be treated differently in Fleets vs Allocations?

Seems like we would want to consolidate on a single implementation (what Allocation does) though if we have that as an option.

Admittedly my initial design only really thought about counter value and list length, not capacity -- but I'm thinking it would be most intuitive for end users for the sorting to be the same?

Reason I ask this here is not to change it here, it's more so I can keep a copy of these changes so that when we do change it, I have the doc updates ready to go, and don't have to write them again.

Reply:
Oh interesting! I didn't expect that, I just assumed they would be the same to be honest! 

I do now have some vague memories of conversations on this topic -- but I can't remember the exact reason we did this.

Something to do with how the Autoscaler works out if it can scale down or not, and matching that logic here so it aligns as close as possible, if I remember correctly? 

Can you remind me again of why this works this way, so I can capture it in the docs and head off any potential confusion here.

Having two different strategies for `priorities` in different places might be confusing (🤔 maybe), but that being said, I always expected the CUJ for this to be more on the count/list length value and capacity to be a non-factor 🤷🏻 .

Reply:
Yeah, I remember talking about it too, but can't find it in chat or meeting notes. 

Since the autoscaler never removes allocated game servers, what this really does is sort the non-allocated game servers, and remove those in order of largest or smallest capacity (after sorting by packed strategy). Which I think we decided was more important than "available capacity" for non-allocated game servers.

Reply:
So I'm digging into the FleetAutoscaler, and I'm seeing that it's also working on available capacity (which makes sense to calculate the buffer).

https://github.com/googleforgames/agones/blob/c35bbf1262f60f7b490524e36cba91c3fdbd4942/pkg/fleetautoscalers/fleetautoscalers.go#L239

As a result of writing the docs, and seeing the different approaches now side by side, I'm thinking we should file a ticket to change the sorting strategy for Fleet scale down to be based on available capacity, and implement it in a separate PR?

9.9 times out of 10, I expect it to be exactly the same as set capacity with a count/length tiebreaker (since I expect dynamic capacities in GameServers that are Ready to be minimal, and counts > 0 in a Ready GameServer to be even less of an occurrence) - but it means that the concepts and the documentation across all functionality align, and are easier to reconcile when working with it?

WDYT @igooch ? If you're happy, I'll file the ticket and make the change - looks like it _should_ be relatively minimal.

Then we can also work out how we want to manage the switchover in docs 😄 but lemme know what you think.

Reply:
The fleet autoscaler buffer is calculated based on available capacity, the priorities are not.

Reply:
Just chatting offline - doesn't seem like a huge concern if I go do the work. So will pause on this doc, and see if it's onerous to change the ordering strategy on Fleet scaledown, and then the documentation is much easier to write.